github/commitstatus.go

```diff
@@ -121,14 +130,29 @@ func (cs CommitStatus) Add(sha, state, targetURL, description string) error {
 	if err != nil {
 		return fmt.Errorf("create http request: %w", err)
 	}
-	req.Header.Set("Authorization", "token "+cs.token)
+	if !cs.useGithubAppAuth {
+		req.Header.Set("Authorization", "token "+cs.token)
+	}
 	req.Header.Set("Accept", "application/vnd.github.v3+json")
 	req.Header.Set("Content-Type", "application/json")
 
 	// The retryable unit of work.
 	workFn := func() error {
 		// By default, there is no timeout, so the call could hang forever.
 		client := &http.Client{Timeout: time.Second * 30}
+		if cs.useGithubAppAuth {
```

Comment:
In my opinion, this is the wrong place to do this... (not the fault of this PR)

Our initial implementation was wrong and we should not initialize the client so low in the chain...